### ISSUE TITLE
UnicodeDecodeError: Non-ASCII characters found in build environment

### ISSUE THREAD

**Issue by ricavalcanti** [platformio/platformio-atom-ide#768]
C:\Users\ricjo\.platformio\packages\tool-scons\script\..\engine\SCons\Environment.py:1303: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal

val = [x for x in val if x not in dk]

C:\Users\ricjo\.platformio\penv\lib\site-packages\platformio\builder\tools\piolib.py:596: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
if path not in self.envorigin.get("CPPPATH", []):

UnicodeDecodeError: Non-ASCII characters found in build environment
See explanation in FAQ > Troubleshooting > Building
http://docs.platformio.org/page/faq.html

 [ERROR] Took 1.89 seconds

**Comment by ivankravets** [comment]
Do you have non-ASCII chars in project folder path?

**Comment by ricavalcanti** [comment]
I fixed that by moving the project folder from google drive folder to desktop

**Comment by ivankravets** [comment]
It should be fixed in PIO Core 3.5.0b2. Could you re-test it?

**Comment by ivankravets** [comment]
Please reopen if you still need help

**Comment by JustACatRu** [comment]
This issue is live again. Kind a.
When using "pio init" with "--ide" parameter (no matter what ide).
I'm talking about PIO CLI install (through "[pip install -U https...](https://docs.platformio.org/en/latest/installation.html#development-version)") last dev version.
Python 3.7.2. All other versions (and error) are visible on screenshot:
![2019-02-01_001338](https://user-images.githubusercontent.com/12075875/52086630-76de4880-25b8-11e9-9375-d56d710a2687.png)
And there is no Non-ASCII characters in my paths as I can see:
![2019-02-01_001404](https://user-images.githubusercontent.com/12075875/52086874-084dba80-25b9-11e9-9737-2feea49bbdc9.png)

**Comment by JustACatRu** [comment]
Forgot to mention: as I understand, this issue does not affect functionality.
I mean, I still can init project and build it without problems (only this error shows up during "init --ide ..." process).
Tested on a project with platform = espressif8266, board = esp12e, framework = arduino.
I will continue testing this bundle (PIO CLI dev + Python 3.7.2 + eclipse as IDE).

**Comment by JustACatRu** [comment]
> this issue does not affect functionality.

I was wrong (because was testing on existing projects). **It is affecting functioality.** Actually "pio init --ide ..." does not create all necessary files and folders and new project could not be imported to IDE after that.
So I need manually create (copy from other project) missing files and folders.

**Comment by JustACatRu** [comment]
I've made some more tests.
This issue related to espressif8266 platform.
If I select some avr board like "pio init --board pro16MHzatmega328" then "pio init --ide eclipse" works without errors:
![2019-02-02_013425](https://user-images.githubusercontent.com/12075875/52153760-4ff14680-268c-11e9-90c1-0411eb1f6dcf.png)
If I select esp board like "pio init --board esp07" then "pio init --ide eclipse" produce error:
![2019-02-02_013619](https://user-images.githubusercontent.com/12075875/52153785-6eefd880-268c-11e9-920c-60bdfebefde6.png)
If there is 2 envs in a project and avr goes first in "platformio.ini" then all is workig fine too:
![2019-02-02_014015](https://user-images.githubusercontent.com/12075875/52153823-90e95b00-268c-11e9-892d-9ae590109e9c.png)
But if I put "env:esp07" first, "pio init --ide eclipse" stops working again.

@ivankravets, should I transfer this issue to some other branch? As it is not "platformio-atom-ide" related.

**Comment by JustACatRu** [comment]
I've modified code a little bit and found traceback of error ([this catch](https://github.com/platformio/platformio-core/blob/e23ca2c10992f10b1906ad616248aa2ef999db23/platformio/builder/main.py#L213) hides traceback):
![2019-02-02_023330](https://user-images.githubusercontent.com/12075875/52155898-21786900-2696-11e9-808f-45d34fa53c4e.png)
Here: https://github.com/platformio/platformio-core/blob/e23ca2c10992f10b1906ad616248aa2ef999db23/platformio/util.py#L420
For me it starts working, if I change that line to any of this variants:
```
result[k] = str(result[k])
result[k] = str(result[k], "windows-1251")
result[k] = result[k].decode("windows-1251")
```
![2019-02-02_024315](https://user-images.githubusercontent.com/12075875/52155980-892eb400-2696-11e9-8b8c-505f3285e08a.png)
I do not know if it's a right solution. And which variant is the best and correct. And why it is working with "windows-1251" encoding. My Python knowledge is way too low for this :)
But hope all this will help solve the problem quicker.

**Comment by ivankravets** [comment]
We have just released a new version [2.7.0](https://github.com/platformio/platformio-atom-ide/releases/tag/v2.7.0), please upgrade.

**Please note that we HIGHLY recommend moving to [PlatformIO IDE for VSCode](https://platformio.org/platformio-ide) which resolves tons of issues and brings a lot of new features.**
